### PR TITLE
Add two new functions: ebirdfreq and ebirdregioncheck

### PR DESCRIPTION
**Pull request by sebpardo**
@ateucher and I were both working on similar functions so we decided to join our efforts are create these two functions for `rebird`.

`ebirdfreq` queries the eBird website (not the API) to download the historical frequency of observation data for any given, hotspot, county, or state/province. This function returns data in long format (just like all other rebird functions) or wide format, which is particularly useful if wanting to create a graph with it.

`ebirdregioncheck` queries eBird's API to determine whether a country, state/province, or country code is valid, and returns a logical value.

We have created documentation, tests, and vignette entries for both functions, and our fork of the package passes `devtools::check`.

### CODE REVIEW

**Comment by sckott** [comment]
thanks @sebpardo - looking it over

File: R/ebirdregioncheck.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  } else if (loctype == "states") {
+    args <- list(rtype = "subnational1", countryCode = substr(loc, 1, 2)) 
+  } else if (loctype == "counties") {
+    args <- list(rtype = "subnational2",
+                 subnational1Code = substr(loc, 1, 5)) 
+  } else {
+    stop("loctype must be one of 'country', 'states', 'counties'")
+  }
+  
+  r <- GET(url, query = args)
```
Comment (sckott) [inline R/ebirdregioncheck.R]: @sebpardo can you use the function `ebase()` ? it's in the `zzz.R` file - trying to stick with that so that if ebird updates the base url/api version we just need to change the URL in one place

Comment (sckott) [inline R/ebirdregioncheck.R]: @sebpardo also add a `...` parameter to this and the other function you added to allow passing in curl options, e.g, see   https://github.com/ropensci/rebird/blob/master/R/ebirdgeo.R#L31

**Comment by sckott** [comment]
@sebpardo @ateucher  looks great, I had a few requests before merging this - you can commit those changes to your the same branch `ateucher:frequency-data`

**Comment by ateucher** [comment]
Awesome, thanks @sckott. @sebpardo do you want to do these? Also, we should add @sebpardo  as a contributor in `DESCRIPTION`. :)

**Comment by sckott** [comment]
> we should add @sebpardo as a contributor in `DESCRIPTION`

sounds good

**Comment by sebpardo** [comment]
@ateucher @sckott I'll do them. Will have them ready by tomorrow morning.

**Comment by ateucher** [comment]
Hey @sebpardo, it looks like you did the fixes @sckott asked for eh? I've added you as an author, so if it looks ok with you Scott, this should be good to merge?

I was going to add `skip_on_cran()` to the new tests but noticed none have them in this package, so I'll open a new issue for that.

**Comment by sebpardo** [comment]
@ateucher yep, they are done and tested. And thanks for adding me! :smile:

**Comment by sckott** [comment]
great, thanks very much!

**Comment by ateucher** [comment]
Sweet, thanks @sckott. Nice work @sebpardo!
